File: PHP/Cap04/c04ex06.php
<?php
// c04ex06.php

printf("CALCULADORA\n\n");

printf("Entre o 1o. valor numerico: ");
$A = floatval(trim(fgets(STDIN)));

printf("Entre o 2o. valor numerico: ");
$B = floatval(trim(fgets(STDIN)));

printf("\n");
printf("Escolha uma opcao de menu:\n");
printf("\n");
printf("[1] - Adicao\n");
printf("[2] - Subtracao\n");
printf("[3] - Multiplicacao\n");
printf("[4] - Divisao\n");
printf("\n");

printf("==> ");
$Opcao = intval(trim(fgets(STDIN)));

printf("\n");

switch ($Opcao) {
  case 1:
    printf("Calculo de adicao\n");
    printf("Resultado: %.2f\n", $A + $B);
    break;
  case 2:
    printf("Calculo de subtracao\n");
    printf("Resultado: %.2f\n", $A - $B);
    break;
  case 3:
    printf("Calculo de multiplicacao\n");
    printf("Resultado: %.2f\n", $A * $B);
    break;
  case 4:
    printf("Calculo de divisao\n");
    if ($B == 0)
      printf("Resultado: ERRO\n");
    else
      printf("Resultado: %.2f\n", $A / $B);
    break;
  default:
    printf("Entrada invalida\n");
    printf("Tente novamente em outro momento\n");
    break;
}

printf("\n");
printf("Tecle <Enter> para encerrar... ");
trim(fgets(STDIN));

?>

File: SCALA/Cap04/c04ex06.scala
// c04ex06.scala

import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    var A: Float = 0.0f
    var B: Float = 0.0f
    var Opcao: Int = 0

    print("Entre o 1o. valor numérico: ")
    A = readFloat()

    print("Entre o 2o. valor numérico: ")
    B = readFloat()

    println()
    println("Escolha uma opção de menu:")
    println()
    println("[1] - Adição")
    println("[2] - Subtração")
    println("[3] - Multiplicação")
    println("[4] - Divisão")
    println()

    print("==> ")
    Opcao = readInt()
    println()

    Opcao match {
      case 1 =>
        println("Cálculo de adição")
        printf("Resultado: %.2f\n", A + B)
      case 2 =>
        println("Cálculo de subtração")
        printf("Resultado: %.2f\n", A - B)
      case 3 =>
        println("Cálculo de multiplicação")
        printf("Resultado: %.2f\n", A * B)
      case 4 =>
        println("Cálculo de divisão")
        if (B == 0) {
          println("Resultado: ERRO")
        } else {
          printf("Resultado: %.2f\n", A / B)
        }
      case _ =>
        println("Entrada inválida")
        println("Tente novamente em outro momento")
    }

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}
